src/pages/index.js

```diff
@@ -2,51 +2,137 @@ import "./index.css";
 
 const profileModalHandler = new PopupWithForm("#profile-edit-modal", (data) => {
-  profileInfo.setUserInfo(data)
+  profileModalHandler.changeButtonText(true);
+  api.updateProfileInfo(data)
+  .then(json => {
+    profileInfo.setUserInfo(json);
+  })
+  .catch(err => {
+    console.error(err);
+  })
+  .finally(() => {
+    profileModalHandler.changeButtonText(false);
+  })
 });
 profileModalHandler.setEventListeners();
 
-const cardModalHandler = new PopupWithForm("#add-card-modal", addCard);
+const avatarModalHandler = new PopupWithForm("#avatar-edit-modal", (data) => {
+  avatarModalHandler.changeButtonText(true);
+  api.updateAvatar(data)
+  .then(json => {
+    api.updateAvatar(json);
+    profileInfo.setAvatar(json);
+  })
+  .catch(err => {
+    console.error(err);
+  })
+  .finally(() => {
+    avatarModalHandler.changeButtonText(false);
+  })
+});
+avatarModalHandler.setEventListeners();
+
+const cardModalHandler = new PopupWithForm("#add-card-modal",
+  data => {
+    cardModalHandler.changeButtonText(true);
+    api.addCard(data)
+    .then(json => {
+      addCard(json);
+    })
+    .catch(err => {
+      console.error(err);
+    })
+    .finally(() => {
+      cardModalHandler.changeButtonText(false);
+    })
+  }
+);
 cardModalHandler.setEventListeners();
 
-//Render Initial cards onto the page
-const gridHandler = new Section({
-  items: initialCards,
+//api for handling http requests
+const api = new Api();
+
+const section = new Section({
+  items: [],
   renderer: addCard
-}, ".cards__grid");
+}, ".cards__grid")
 
-gridHandler.renderItems();
+//used to render initial cards
+function addCard (data) {
+  const card = new Card(data, "#card-template",
+  event => {  //image handler
+    imageModalHandler.open(event);
+  },
+  (data, event) => {  // delete button handler
+    deleteModalHandler.open();
+    deleteModalHandler.setSubmitAction(() => {
+      api.removeCard(data)
+      .then(json => {
+        card.deleteCard(event);
+      })
+      .catch(err => {
+        console.error(err);
+      })
+    })
+  },
+  (data, event) => {  //like button handler
+    if (!card.liked) {
+      api.addLike(data)
+      .then(json => {
+        card.updateLikeCount(json);
+        card.toggleLikeButton(event);
+        card.liked = !card.liked;
+      })
+      .catch(err => {
+        console.error(err);
+      })
+    } else {
+        api.removeLike(data)
+        .then(json => {
+          card.updateLikeCount(json);
+          card.toggleLikeButton(event);
+          card.liked = !card.liked;
+        })
+        .catch(err => {
+          console.error(err);
+        })
+    }    
+  }
+  )
+  section.addItem(card.returnCard());
+}
+
+const profileInfo = new UserInfo({
+  nameSelector: ".profile__user-name",
+  aboutSelector: ".profile__about",
+  avatarSelector: ".avatar"
+});
+
+api.getUserAndCardInfo().then(
+  data => {
+    profileInfo.setUserInfo(data[0]);
+    section.items = data[1];
+    section.renderItems();
+  });
```

Comment:
You need to `catch` possible errors at the very end of any server request.

**Please, correct it everywhere**